lib/std/crypto/aes.zig

```diff
@@ -215,6 +240,29 @@ fn AESDecrypt(comptime keysize: usize) type {
         pub fn decrypt(ctx: Self, dst: []u8, src: []const u8) void {
             decryptBlock(ctx.dec[0..], dst, src);
         }
+
+        pub fn ige(ctx: Self, dst: []u8, src: []const u8, iv: [(keysize / 8) * 2]u8) void {
+            std.debug.assert(src.len >= dst.len);
+            
+            const b = keysize / 8;
+            var c: [b]u8 = undefined;
+            var m: [b]u8 = undefined;
+
+            std.mem.copy(u8, &c, iv[0..b]);
+            std.mem.copy(u8, &m, iv[b..]);
+
+            var o: usize = 0;
+            while (o < src.len) : (o += b) {
+                const t = src[o..o+b];
+
+                _ = xorBytes(dst[o..o+b], src[o..o+b], &m);
+                ctx.decrypt(dst[o..o+b], dst[o..o+b]);
+                _ = xorBytes(dst[o..o+b], dst[o..o+b], &c);
+
+                std.mem.copy(u8, &m, dst[o..o+b]);
+                std.mem.copy(u8, &c, t);
+            }
+        }
```

Comment:
what happens if `src.len` is not divisible by `b`?
either need handling for the tail; or an assert + doc comment.